pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -179,8 +181,30 @@ func (c *client) UpdateFunction(ctx context.Context, fm FunctionManifest) error
 		return fmt.Errorf("unknown error given: %w", err)
 	}
 
-	// TODO: Support more configurable fields using Lambda.UpdateFunctionConfiguration.
-	// https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#UpdateFunctionConfiguration
+	retry := backoff.NewRetry(RequestRetryTime, backoff.NewConstant(RetryIntervalDuration))
+	updateFunctionConfigurationSucceed := false
+	var configErr error
+	for retry.WaitNext(ctx) {
+		configInput := &lambda.UpdateFunctionConfigurationInput{
+			FunctionName: aws.String(fm.Spec.Name),
+			MemorySize:   aws.Int64(fm.Spec.Memory),
```

Comment:
Set this value only if Spec.Memory is not zero.

Reply:
You're right 👍 But I have a point to discuss here. Suppose users changed the Memory value once in the past, then for the next time they remove this Memory value field from function.yaml file (the same with Timeout). In that case, which value of memory (and timeout) should be used to apply to the new version of this Lambda application? The previous values or the default values (128 for memory and 1 for timeout). 🤔

Reply:
Note that in case `memory` and `timeout` value is not provided, errors will be raised as bellow
```

Failed to update lambda function SimpleFunction: failed to update configuration for Lambda function SimpleFunction: InvalidParameter: 2 validation error(s) found. - minimum field value of 128, UpdateFunctionConfigurationInput.MemorySize. - minimum field value of 1, UpdateFunctionConfigurationInput.Timeout. 
```
And users can know what should they do to fix it currently 😄

Reply:
@nghialv To keep the simplicity, I think to make `Timeout` and `Memory` as required fields on function.yaml is good enough. What do you think? 🤔 
Addressed necessary changes on https://github.com/pipe-cd/pipe/pull/1548/commits/52f6f5caef804e23d9f89d5d3d4acca36b02e081

Reply:
That looks good too. 👍

Reply:
Thanks 😁  
Update tests to cover those cases on https://github.com/pipe-cd/pipe/pull/1548/commits/3433a846eb1d20bbb00abf9e5a80863d3d3e9539
PTAL 🙏